options.go

```diff
@@ -220,8 +225,18 @@ func DisableRelay() Option {
 	}
 }
 
-// EnableAutoRelay configures libp2p to enable autorelay advertising; requires relay to
-// be enabled and the Routing option to provide an instance of ContentRouting.
+// EnableAutoRelay configures libp2p to enable the AutoRelay subsystem. It is an
+// error to enable AutoRelay without enabling relay (enabled by default) and
+// routing (not enabled by default).
+//
+// This subsystem performs two functions:
+//
+// 1. When this libp2p node is configured to act as a relay "hop"
+//    (circuit.OptHop is passed to EnableRelay), this node will advertise itself
+//    as a public relay using the provided routing system.
+// 2. When this libp2p node is _not_ configured as a relay "hop", it will
+//    automatically if it is unreachable (e.g., behind a NAT). If so, it will
+//    find, configure, and announce a set of public relays.
```

Comment:
@vyzo, this is why I don't like this option design. I now have to explain to the user that the "AutoRelay" option can do two _very_ different things depending on how they've configured their relay.

Reply:
AutoRelay is a system service. 
If you are a hop relay, then the functionality is to advertise relay services. If you are a client, then the functionality is to use automatically use the services advertised by hop relays.

Reply:
What service? There are, and always have been, two _completely separate and orthogonal services_: one for "hop" relays and one for clients. That's why this is so damn confusing.

Reply:
I tend to think of `autorelay` as a single service. But yes, at some level you are right that these are two seperate services and that could potentially be confusing.

Reply:
> There are, and always have been, two completely separate and orthogonal services: one for "hop" relays and one for clients.

Actually, it's even more confusing than that. A _client_ can be a _consumer_ of a relay service, or a _target_ of a relayed connection. Currently, all of these flow over the same thick protocol. I really have a hard time wrapping my head around this, and so do our users.

I'd like to segregate these protocols if possible:

* `/libp2p/circuit/relay/hop/0.1.0`: exposed only by relay nodes. Used inside the initiator -> relay connection.
* `/libp2p/circuit/relay/target/0.1.0`: exposed by clients accepting relayed connections. Used in the relay -> target connection.

This would allow for better implicit observability too. Would've helped a lot in the incident we're analysing.